src/main/java/cloud/eppo/EppoClient.java

```diff
@@ -197,8 +197,17 @@ public EppoClient buildAndInit() {
               pollingIntervalMs,
               pollingIntervalMs / DEFAULT_JITTER_INTERVAL_RATIO);
 
-      // Kick off the first fetch
-      fetchConfigurationsTask.run();
+      fetchConfigurationsTask.scheduleNext();
+
+      // Kick off the first fetch, respecting graceful mode.
+      try {
+        instance.loadConfiguration();
```

Comment:
This actually makes the `buildAndInit` method blocking and guarantees the client is ready to make assignments immediately after calling

Reply:
Is this OK to do?

Reply:
If I understand correctly, the old behavior here was to kick off a background thread to perform the init and return immediately, while it now blocks the thread and doesn't return until `loadConfiguration` finished. is that correct?
if so, seems like an API breaking change to me

Reply:
That's correct.
I have to agree on your assessment of this being a breaking change; while it doesn't break the API signature, it breaks the contract of the API (non blocking, not guaranteed to be ready to make assignments).

I've updated to keep the current behaviour in place: clients is not guaranteed to be ready after init (there's no ability to block-until-ready or get a callback), AND, graceful mode is intrinsically true during Init.
We can revisit these with the next major bump.